### PR TITLE
Migrate pyproject

### PR DESCRIPTION
**Pull request by synesthesiam**
Use pyproject.toml instead of setup.py
Change to date versioning
Add parameter for json loader in get_intents

### CODE REVIEW

File: package/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+[build-system]
+requires = ["setuptools~=62.3", "wheel~=0.37.1"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name        = "home-assistant-intents"
+version     = "2022.1.0.dev0"
```
Comment (synesthesiam) [inline package/pyproject.toml:7]: @balloob  Not sure if this is what we want to do or not for versioning.

Comment (balloob) [reply package/pyproject.toml:7]: yeah that works.

File: package/home_assistant_intents/__init__.py [hunk 2]
```diff
@@ -2,28 +2,24 @@
...
-except (ImportError, AttributeError):
-    # Backport for Python < 3.9
-    import importlib_resources  # type: ignore
-
-    files = importlib_resources.files
+import importlib.resources
 
 _PACKAGE = "home_assistant_intents"
-_DIR = Path(typing.cast(os.PathLike, files(_PACKAGE)))
+_DIR = Path(typing.cast(os.PathLike, importlib.resources.files(_PACKAGE)))
```
Comment (balloob) [inline package/home_assistant_intents/__init__.py:10]: Why can't we do `Path(__file__).parent` ?

**Review by balloob — APPROVED** [review]

File: package/home_assistant_intents/__init__.py (hunk 2 above)
Comment (synesthesiam) [reply package/home_assistant_intents/__init__.py:10]: It should allow the package to be zip safe, which I would need to mark in the `pyproject.toml`.

**Pull request merged**